### PR TITLE
Build rootfs from public docker images

### PR DESCRIPTION
**Pull request by klutchell**

### CODE REVIEW

File: docker-compose.test.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: "2.4"
+
+services:
+  sut:
+    build: .
+    image: localhost:5000/sut
+    privileged: true
```
Comment (jakogut) [inline docker-compose.test.yml:7]: Is privileged actually required?

File: entry.sh [hunk 2]
```diff
@@ -2,40 +2,39 @@
...
+
+kernel_path="/files/vmlinux.bin"
+rootfs_path="/files/rootfs.ext4"
+
+mkdir -p /files
+
+set -x
+
+if [ ! -f "${rootfs_path}" ]; then
+    dd if=/dev/zero of="${rootfs_path}" bs=1M count="${ROOTFS_SIZE_MB}"
```
Comment (jakogut) [inline entry.sh]: This doesn't need to be fully allocated, you can do `truncate -s "${ROOTFS_SIZE_MB}" "${rootfs_path}"` instead.

**Review by jakogut — CHANGES_REQUESTED** [review]

File: docker-compose.test.yml (hunk 1 above)
Comment (klutchell) [reply docker-compose.test.yml:7]: Perhaps not anymore, but some permissions will be needed for mounting the ext roots.

File: entry.sh (hunk 2 above)
Comment (klutchell) [reply entry.sh]: Nice!

File: entry.sh [hunk 3]
```diff
@@ -2,40 +2,45 @@
...
+BLOCK_SIZE=4096
+
+mkdir -p /files
+
+# for now always force a new rootfs and kernel download
+rm -fv /files/*
+
+if [ ! -f "${rootfs_path}" ]; then
+    echo "Creating rootfs ${rootfs_path}..."
+    truncate -s "$((ROOTFS_SIZE_MB * 1024 * 1024 / BLOCK_SIZE * BLOCK_SIZE))" "${rootfs_path}"
```
Comment (jakogut) [inline entry.sh]: `/ BLOCK_SIZE * BLOCK_SIZE`

wat

**Review by jakogut — APPROVED** [review]
LGTM

**Pull request merged**